src/components/Footer.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useContext, useMemo } from 'react';
+import { TodoFilter } from './TodoFilter';
+import { TodoContext } from '../context/TodoContext';
+import { Error } from '../types/ErrorMessage';
+import * as TodoService from '../api/todos';
+
+export const Footer: React.FC = () => {
+  const { todos, deleteTodo, handleSetErrorMessage, handleSetUpdatingIds } =
+    useContext(TodoContext);
+
+  const completedIds: number[] = useMemo(
+    () => todos.filter(({ completed }) => completed).map(({ id }) => id),
+    [todos],
+  );
+  const hasActive = useMemo(
```

Comment:
```suggestion
  const activeTodosCount = useMemo(
```
hasActive is a bit misleading name (we use `has`, `is` prefixes with boolean variables)